server/index.js

```diff
@@ -216,12 +215,19 @@ Visit the documentation for more information: https://bpanel.org/docs/configurat
 
     app.get('/', resolveIndex);
 
-    // get default config for fallback endpoint
-    const defaultClientConfig = getDefaultConfig(bpanelConfig);
-    const clientId = defaultClientConfig.str('id');
+    // compose endpoints
+    const apiEndpoints = [];
+    for (let key in endpoints) {
+      apiEndpoints.push(...endpoints[key]);
+    }
 
-    // client routes
-    app.use('/clients', clientRoutes(clients, clientId));
+    for (let endpoint of apiEndpoints) {
+      try {
+        attach({ app, endpoint, logger, bpanelConfig });
```

Comment:
Any reasoning behind using an object + destructuring here?

Reply:
So one reason is it's a habit a picked up from linters that enforce max number of parameters in the signature. Second is it could simplify use of it later on. If we add `logger` and `bpanelConfig` to the `req` object though, then we'd only have two parameters so I think normal signature should be fine after that update.

Reply:
Just was following up with an explanation of my reasoning 👍